subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/ArtifactsTestListener.kt

```diff
@@ -63,12 +63,20 @@ internal class ArtifactsTestListener(
                 }
                 TestResult.Complete(artifacts)
             }
+
             is InfrastructureError ->
-                TestResult.Incomplete(result)
+                TestResult.Incomplete(
+                    infraError = result,
+                    logcat = device.logcat(LOGCAT_LINES_TO_FETCH)
+                )
+
             Ignored ->
                 TestResult.Incomplete(
                     InfrastructureError.Unexpected(
                         IllegalStateException("Instrumentation executed Ignored test")
+                    ),
+                    logcat = Result.Failure(
```

Comment:
It looks that Result is not suitable for this property.
Clients can't distinguish "empty\disabled" values from real errors.

Reply:
Agreed, please take a look at new `LogcatResult`

Reply:
we should, but can't right now, it needs to be refactored in a way that most internal classes of logcat and report processing moved to `:client` module

Reply:
I don't get why we can't call `logcatBuffers.get(key)` when test is incomplete?

Reply:
because logcat buffers needs start and stop events, and such incomplete tests doesn't have these callbacks called